client/me/purchases/manage-purchase/index.jsx

```diff
@@ -46,22 +46,33 @@ import {
 	showCreditCardExpiringWarning,
 	showEditPaymentDetails
 } from 'lib/purchases';
-import { getPurchase, goToList, isDataLoading, recordPageView } from '../utils';
+import { getPurchase, getSelectedSite, goToList, isDataLoading, recordPageView } from '../utils';
 import { isDomainProduct, isGoogleApps, isPlan, isSiteRedirect, isTheme } from 'lib/products-values';
 
 const ManagePurchase = React.createClass( {
 	propTypes: {
 		cart: React.PropTypes.object.isRequired,
 		selectedPurchase: React.PropTypes.object.isRequired,
-		selectedSite: React.PropTypes.object,
+		selectedSite: React.PropTypes.oneOfType( [
+			React.PropTypes.object,
+			React.PropTypes.bool
+		] ).isRequired,
 		destinationType: React.PropTypes.string
 	},
 
 	componentWillMount() {
+		if ( ! this.isDataValid() ) {
+			return page.redirect( paths.list() );
```

Comment:
> I admit I don't know fully whats going on around and outside of this component but redirecting after the component has mounted because of missing props seems like this could benefit from some rethinking. I think the redirect should happen a bit higher in the stack, preferably the controller.

I agree that this could be moved one level up to `<StoreConnection />` component. However in the case of `Manage Purchase` page this would require also extracting loading placeholder code into it's own component together with all it's dependent logic. Let's leave it as it is for now, and revisit when we are done with all page redirects planned in #275. We also need to keep in mind that some Purchases pages share Data components.